src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -45,146 +73,1195 @@ ActorInit En_Slime_InitVars = {
+        this->actor.velocity.y = 0.0f;
+    }
+    this->collider.base.acFlags &= ~AC_ON;
+    this->timer = 30;
+    Actor_PlaySfx(&this->actor, NA_SE_EN_SLIME_DEFENCE);
+    if (this->collider.base.ac != NULL) {
+        this->actor.world.rot.y = Actor_WorldYawTowardActor(&this->actor, this->collider.base.ac) + 0x8000;
+    } else {
+        this->actor.world.rot.y = this->actor.yawTowardsPlayer + 0x8000;
+    }
+    this->actionFunc = EnSlime_ReactToBluntHit;
+}
 
-extern ColliderCylinderInit D_80A31AF0;
-extern DamageTable D_80A31B1C;
-extern CollisionCheckInfoInit D_80A31B3C;
-extern InitChainEntry D_80A31B54[];
+/**
+ * Update reaction to receiving a blunt hit (such as with Goron Punch etc.)
+ * Update XZ speed and jiggle effect
+ * AC_ON flag is off until 15 frames of reaction remain.
+ * When timer is up, chuchu will choose whether to engage player, return home,
+ * or return to idle.
+ * If chuchu is too far from home, returning home will take precedence.
+ */
+void EnSlime_ReactToBluntHit(EnSlime* this, PlayState* play) {
+    f32 timerFactor;
+    f32 scale;
+
+    EnSlime_Blink(this);
+    this->timer--;
+    Math_StepToF(&this->actor.speed, 0.0f, 1.0f);
+    timerFactor = sqrtf(this->timer);
+    if (this->timer < 30) {
+        scale = ((cos_rad(this->timer * (2.0f * M_PI / 5.0f)) * (0.08f * timerFactor)) + 1.0f) * 0.01f;
+        this->actor.scale.x = scale;
+        this->actor.scale.z = scale;
+    }
+    if (this->timer == 15) {
+        this->collider.base.acFlags |= AC_ON;
+    }
+    this->actor.scale.y = ((sin_rad((f32)this->timer * (2.0f * M_PI / 5.0f)) * (0.07f * timerFactor)) + 1.0f) * 0.01f;
+    this->actor.shape.rot.x = randPlusMinusPoint5Scaled(0x200) * timerFactor;
+    this->actor.shape.rot.z = randPlusMinusPoint5Scaled(0x200) * timerFactor;
+
+    if (this->timer == 0) {
+        if (Actor_WorldDistXZToPoint(&this->actor, &this->actor.home.pos) > this->distLimit) {
+            EnSlime_SetupMoveToHome(this);
+        } else if ((this->actor.xzDistToPlayer < 280.0f) && (Player_GetMask(play) != PLAYER_MASK_STONE)) {
+            EnSlime_SetupTurnToPlayer(this);
+        } else {
+            EnSlime_SetupIdle(this);
+        }
+    }
+}
+
+/**
+ * Initialize reaction to taking damage.
+ * Spawn bubbles & splash, start actor rebuff wobble, push actor, and play sfx.
+ */
+void EnSlime_SetupDamaged(EnSlime* this, PlayState* play, s32 faceDamageSource) {
+    s32 i;
+    Vec3f effectVelocity;
+    Vec3f effectPos;
+    f32 ySin;
+    f32 yCos;
+
+    this->collider.base.acFlags &= ~AC_ON;
+    if (this->actor.velocity.y > 0.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0x2000, 20);
+    this->timer = 20;
+    this->actor.speed = 10.0f;
+    if (faceDamageSource == true) {
+        // This function changes actor's world Y rotation - I think it's maybe to face source of damage?
+        func_800BE504(&this->actor, &this->collider);
```

Comment:
Hmm, I tested this in-game, and it doesn't seem be affected too much:
https://user-images.githubusercontent.com/12245827/226514312-70a86a9d-6c9f-4820-b00a-266c3a074701.mp4
https://user-images.githubusercontent.com/12245827/226513730-a9365ef7-6a25-4858-b501-a929fee8a414.mp4
https://user-images.githubusercontent.com/12245827/226514339-83cb33f2-5efc-4b28-afd0-c842e1ac585f.mp4

I wonder if we should just leave this as `arg1`, since we don't really know what it does; that's what a few actors (like EnIk and EnWallmas) do.

Reply:
I guess it would technically be arg2 in this case? Yeah, I'll go ahead and change it to that, then.